conda_forge_tick/utils.py

```diff
@@ -667,6 +715,42 @@ def _parse_validated_recipes(
     return _remove_none_values(parsed_recipes)
 
 
+def parse_recipe_yaml_requirements(requirements):
+    """Parse requirement section of render by rattler-build to fit `RecipeTypedDict`
+
+
+    When rendering the recipe by rattler build,
+    `requirements["run_exports"]["weak"]` gives a list looking like:
+    [
+        {
+          "pin_subpackage": {
+            "name": "slepc",
+            "lower_bound": "x.x.x.x.x.x",
+            "upper_bound": "x.x"
+          }
+        }
+    ]
+    `run_exports["weak"]` of RecipeTypedDict looks like:
+    [
+        "pin_subpackage"
+    ]
+
+    The same applies to "strong".
+
+    This function takes care of this transformation
+    """
+    if run_exports := requirements.get("run_exports"):
+        weak = run_exports.get("weak")
+        if isinstance(weak, list) and isinstance(weak[0], dict):
+            run_exports["weak"] = [value for entry in weak for value in entry.values()]
+
+        strong = run_exports.get("strong")
+        if isinstance(strong, list) and isinstance(strong[0], dict):
+            run_exports["strong"] = [
+                value for entry in strong for value in entry.values()
```

Comment:
I would have expected `.keys()` here (and above) given the doc string? I also think this might interact weirdly with the parsing for pinning since it comes after and is undoing dicts.

Let's add a unit test for this function to ensure it works properly and help document it's behavior.

Reply:
I've added three